xwiki-platform-core/xwiki-platform-panels/xwiki-platform-panels-ui/src/main/resources/Panels/OrphanedPages.xml

```diff
@@ -36,7 +36,41 @@
   <minorEdit>false</minorEdit>
   <syntaxId>xwiki/2.0</syntaxId>
   <hidden>true</hidden>
-  <content/>
+  <content>{{include reference="XWiki.DocumentTreeMacros" /}}
+
+{{velocity wiki='false'}}
+#if ($xcontext.action == 'get')
+  #if ($request.offset)
+    #set ($offset = $request.offset)
+  #else
+    #set ($offset = 0)
+  #end
+
+  #if ($request.limit)
+    #set ($limit = $request.limit)
+  #else
+    #set ($limit = 20)
+  #end
+
+  #set ($homepage = $services.wiki.getById($services.wiki.currentWikiId).mainPageReference)
+  #set ($homepageFullName = $services.model.serialize($homepage, 'local'))
+
+  #set ($orphanedItem = $services.panels.listOrphaned($limit, $offset, $homepageFullName))
+
+  #set($data = [])
+  #updateDocTreeConfigFromRequest()
+  #foreach ($item in $orphanedItem.orphanedPages)
+    #addDocumentNode($item $data)
+  #end
+  #if ($orphanedItem.getHasMore())
+    #addPaginationNode($homepageFullName $orphanedItem.offset '' $data)
+    ## Replaces the pagination node with a generic message since we do not count the total number of orphaned pages.
```

Comment:
Is there a problem to count the total number of orphan pages? I mean, couldn't we replace ``hasMore`` with ``totalCount``?